test/nodes/test_prompt_node.py

```diff
@@ -11,8 +11,26 @@
 from haystack.nodes.prompt.prompt_node import HFLocalInvocationLayer
 
 
-def is_openai_api_key_set(api_key: str):
-    return len(api_key) > 0 and api_key != "KEY_NOT_FOUND"
+def skip_test_for_invalid_key(prompt_model):
+    if prompt_model.api_key is not None and prompt_model.api_key == "KEY_NOT_FOUND":
+        pytest.skip("No API key found, skipping test")
+
+
+def get_either_openai_or_azure_openai_api_key():
+    openai_api_key = os.environ.get("OPENAI_API_KEY", None)
+    azure_openai_api_key = os.environ.get("AZURE_OPENAI_API_KEY", None)
+    if azure_openai_api_key:
+        return azure_openai_api_key
+    elif openai_api_key:
+        return openai_api_key
+    else:
+        return None
+
+
+either_openai_or_azure_openai_api_key_in_env = pytest.mark.skipif(
```

Comment:
Same as this [comment](https://github.com/deepset-ai/haystack/pull/4170/files#r1121785484). I think this should be removed and we should directly specify which API we are using in the test.